modules/ROOT/pages/keycloak/authenticating.adoc

```diff
@@ -51,19 +48,180 @@ public void onActivityResult(int requestCode, int resultCode, Intent data) {
         authService.handleAuthResult(data);
     }
 }
+----
++
+NOTE: This handler should always invoke `handleAuthResult`, providing the `Intent`. This will exchange the temporary tokens returned from `login` for long-lived tokens and will provide a `UserPrincipal` which can be used to access a users details. If this is not invoked you will not have access to the `UserPrincipal`.
+
+. The callback provided to `login` will be invoked.
+****
+
+[role="secondary"]
+.iOS
+****
+. Use the `login` function to trigger the authentication flow:
++
+[source,swift]
+----
+try AgsAuth.instance.login(presentingViewController: self, onCompleted: onLoginComplete)
+
+// handle the authentication result
+func onLoginComplete(user: User?, err: Error?) {
+    // login failed
+    if let error = err {
+        // login error
+        return
+    }
+    // login succeeded
+    let currentUser = user
+}
+----
+
+. Allow your application to handle the redirect in `AppDelegate`:
++
+[source,swift]
+----
+func application(_ app: UIApplication, open url: URL, options: [UIApplicationOpenURLOptionsKey: Any] = [:]) -> Bool {
+    do {
+        return try AgsAuth.instance.resumeAuth(url: url as URL)
+    } catch AgsAuth.Errors.serviceNotConfigured {
+        print("AeroGear auth service is not configured")
+    } catch {
+        fatalError("Unexpected error: \(error).")
+    }
+    return false
+}
+----
+****
+
+[role="secondary"]
+.Cordova
+****
+Authentication will be performed on initialization as described xref:keycloak/coding.adoc[here]. To manually redirect to the login screen use:
+[code,javascript]
+----
+authService.login()
+----
+****
+
+[role="secondary"]
+.Xamarin
+****
+. Get an instance of `IAuthService`:
```

Comment:
Again let's be consistent. This step of _getting an instance_ is not explicit in the other platforms. I am in favor of having this step, but then let's add it to Android, iOS and Cordova.